docs/btc-transactions-validity.md

```diff
@@ -1,23 +1,179 @@
+commits these conditions within its Taproot structure.
+
+#### Building a Staking Transaction Using the Babylon Library
+
+To simplify the creation of a staking transaction the Babylon staking library provides its own function.
+
+##### Function Overview
+
+The Babylon staking library exposes the
+[BuildV0IdentifiableStakingOutputsAndTx](https://github.com/babylonlabs-io/babylon/blob/main/btcstaking/identifiable_staking.go?plain=1#L231)
+function, which is used to construct the staking transaction. This function
+generates the following:
+
+- A Taproot UTXO (locks BTC using `Q`)
+- An `OP_RETURN` output, storing the staking metadata
+
+```go
+func BuildV0IdentifiableStakingOutputsAndTx(
+ tag []byte,
+ stakerKey *btcec.PublicKey,
+ fpKey *btcec.PublicKey,
+ covenantKeys []*btcec.PublicKey,
+ covenantQuorum uint32,
+ stakingTime uint16,
+ stakingAmount btcutil.Amount,
+ net *chaincfg.Params,
+) (*IdentifiableStakingInfo, *wire.MsgTx, error)
+```
+
+To construct the full staking transaction, the user must provide the following
+parameters:
+
+**Parameters**:
+
+- `Tag` - 4 bytes, a tag which is used to identify the staking transaction
+    among other transactions in the Bitcoin ledger. It is specified in the
+    `global_parameters.Tag` field.
+- `Version` - 1 byte, the current version of the OP_RETURN output.
+- `StakerPublicKey` - 32 bytes, staker public key. The same key must be used
+    in the scripts used to create the Taproot output in the staking transaction.
+- `FinalityProviderPublicKey` - 32 bytes, finality provider public key. The
+    same key must be used in the scripts used to create the Taproot output in
+    the staking transaction.
+- `StakingTime` - 2 bytes big-endian unsigned number, staking time. The same
+    timelock time must be used in scripts used to create the Taproot output in
+    the staking transaction.
+
+**Where does this data go?**
+
+For the full transaction, the provided parameters (listed above) are used to
+construct two main outputs:
+
+1. The Taproot UTXO (`staking_output`), which enforces the staking rules on
+    Bitcoin.
+2. The `OP_RETURN` output, which stores Babylon-specific metadata.
+
+This function internally calls `BuildV0IdentifiableStakingOutputs()`, which:
+    - Retrieves `global_parameters`.
+    - Generates the OP_RETURN output using the parameters above.
+    - Constructs the Taproot UTXO, embedding the staking conditions inside the
+        Merkle root.
+
+##### OP_RETURN Output
+
+To store staking metadata on-chain, Babylon uses an OP_RETURN output. This
+output is built using the `V0OpReturnData` struct:
+
+```go
+type V0OpReturnData struct {
+ Tag                       []byte
+ Version                   byte
+ StakerPublicKey           *XonlyPubKey
+ FinalityProviderPublicKey *XonlyPubKey
+ StakingTime               uint16
+}
+
+```
+
+<!-- should we still include serialisation format for OP_RETURN data? -->
```

Comment:
I think I would leave out the op_return at all from this docs as it is inteneded for phase-2 and onwards.